### ISSUE TITLE
High Sierra

### ISSUE THREAD

**Issue by pedrohex** [Floris497/mac-pixel-clock-patch-V2#222]
Hi, 
The patch was working great until high sierra update.
I've tried V1, V2, V3 and V4 with no results.. with V4 mac get stuck at startup and i must unpatch it.

MacbookPro 13 early 2011
Using Apple HDMI to Minidisplay port
Monitor LG UltraWide 29UM59-P 2560x1080

Any ideas ? 
Thanks in advance

**Comment by Floris497** [comment]
Nope, no idea yet.. v4 isn't ready

**Comment by pedrohex** [comment]
Thank you for your response Floris497.
Please tell me if i can help some how.

Meanwhile , downgrading to sierra :(

**Comment by gothickitty93** [comment]
just out of curiosity, is there a major difference between coredisplay in high sierra vs sierra due to more things running on Metal?

**Comment by Floris497** [comment]
I remember the executible is serveral MB bigger.. not sure if that has to do with metal..

**Comment by Bastani** [comment]
**Make sure you unpatch before using any of the following:**
You can either use this script which will unfortunately break Copy and Paste:
`sudo perl -i.bak -pe 's|\xB8\x01\x00\x00\x00\xF6\xC1\x01\x0F\x85|\x33\xC0\x90\x90\x90\x90\x90\x90\x90\xE9|sg' /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay
sudo codesign -f -s - /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay`

Or [Lilu](https://github.com/vit9696/Lilu) + [CoreDisplayFixup](https://github.com/PMheart/CoreDisplayFixup) which you have to compile yourself but should not have the copy and paste problem. 

Both of the above have an issue with iTunes crashing. No idea how to fix it.

Credit to [scottsanett](https://github.com/Floris497/mac-pixel-clock-patch-V2/issues/211#issuecomment-327346109) for the above script

**Comment by Floris497** [comment]
I was trying core displayfixup and lulu.. but I couldn't see if it was working or not.. maybe it's the future :)

**Comment by yverbin** [comment]
try it:
`sudo perl -i.bak -pe 's|\xB8\x01\x00\x00\x00\xF6\xC1\x01\x0F\x85|\x33\xC0\x90\x90\x90\x90\x90\x90\x90\xE9|sg' /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay
sudo codesign --deep -f -s - /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay`

it works fine for me (macos ver. 10.12.6)

**Comment by Bastani** [comment]
Yeah coredisplayfixup + Lulu worked perfectly for me, I could upload the kexts I compiled if you like.

**Comment by Floris497** [comment]
Precompiled kexts are available already on github I'll link to it :) it seems to be a more permanent fix for this problem :) I think everybody should move that way! I'll edit the readme with some links tomorrow

**Comment by Bastani** [comment]
Actually you'll need Lilu 1.2 and CoreDisplayFixup 1.3 which neither have been released yet. Everyone would have to compile it

**Comment by Floris497** [comment]
Ah I wasn't aware.. you can post them :) and later we'll link somewhere permanent

**Comment by Bastani** [comment]
Here I have provided a link to my compiled versions of the Kexts: [Lulu + CoreDisplayFixup](https://drive.google.com/open?id=0B2m8njhk5gkOZzNoNUtqZ3kwajQ)

**Comment by cmauribon** [comment]
Hello, I'm sorry but i need help... I've an hd5500 with 4k screen, lilu and coredisplayfixup put in clover/kext/other, but my laptop reboot so fast i cannot see anything... I think i'm missing something...

**Comment by funlen** [comment]
Thank you Bastani, your build is working fine on my 10.13,  my 2K monitor can output 2560*1440@60hz now.

**Comment by spencerjason** [comment]
Can someone explain how I use the Lulu + CDF kext files? My MacMini (2012) is sad on my LG UltraWide :/

**Comment by spencerjason** [comment]
Nevermind, I found the correct terminal procedure. Although I didn't seem to have a cache to remove, which seemed off (no such files existed). Reboot let me pick 2560x1080 @60hz with no issues.

Fingers crossed it holds up. Thanks @Bastani for the compiled link. 👍

**Comment by LETRA** [comment]
@spencerjason Please explain the entire process

**Comment by Floris497** [comment]
Tonight when I'm home I'll rewrite the readme after some testing :)

**Comment by HubertPluskota** [comment]
@spencerjason Hey, can you give the command you used?

**Comment by cmauribon** [comment]
@spencerjason : how do you use lilu + CDF ? I can't get it work in clover, I also remove cache, without change, bootloop :(

**Comment by spencerjason** [comment]
Not sure I know the exact steps to replicate, and I don't know anything about clover, however when I tried to install Pixel-Clockv2/3/4 it gave all kinds of errors when trying to patch. In the process, I lost the ability to cut-n-paste and my .bak was missing. So I bit the bullet, rebooted into recovery and reinstalled the OS (kept existing files, so not fresh install).

With that, I followed [these directions](http://osxdaily.com/2012/01/12/how-to-manually-install-kernel-extensions-in-mac-os-x/) for installing both Lilu and CDF .kext files.

NOTE: Whenever I tried to remove the kext caches I was told the files didn't exist. I rebooted anyway and when I went to the display dialogue, the new resolutions were already there. I still had to switch to it for it to use, but I've not had any issues in the last 24 hours that I've been running.

I'm sure that @Floris497 or @Bastani will be better on the technical side of describing it when they have time to update.

**Comment by cmauribon** [comment]
@spencerjason Lilu and CDF not working for me from a clean high sierra install... In verbose mode,it's boot normaly but when it go to load framebuffer there is a lot of text and It reboot so fast i cannot take a picture on what it going on

**Comment by TylerD89** [comment]
@cmauribon Kernel panics don't really point out much info on what causes them. In fact, all it gives you is text that doesn't really make any sense. If you are using Clover, press space on the macOS boot option, scroll down with arrow keys and find "Don't Reboot on Panic" and press space on it, scroll back up to the top and press enter on "Boot macOS with selected boot options"

**Comment by cmauribon** [comment]
There is a lot of display's error/bug with High sierra, maybe we should wait for a more stable version

**Comment by juvenal** [comment]
@cmauribon, the final version on macOS 10.13 (High Sierra) was released by Apple on September, 25th... I think Apple wouldn't release anything "more stable" to wait for...

**Comment by cmauribon** [comment]
@juvenal : I saw multiple user complaining with graphical issues with external display on macos 10.13.0

**Comment by cmauribon** [comment]
and apfs is really unstable ( on hackintosh ) : dirty reboot could result on a unreadable partition

**Comment by Bastani** [comment]
The issues you're having will likely not get fixed by apple as they have nothing to do with their end, it's just something the hackintosh community has to figure out how to fix. Also you do realize that It's not only people with a hackintosh that uses this patch right?

**Comment by Floris497** [comment]
This thing was never meant for hackintosh actually, it came to be.. i think now the best way ti patch is with lilu, I've not tried it.. i have no compatible hardware running anymore.. so sorry for all.. it's getting mor difficult for me to maintain :/

**Comment by Bastani** [comment]
Yeah thank you for all your hard work @Floris497, It was good while it lasted, people here will just have to try and get it to work with Lilu, It seem to be the way forward. I am personally using it with a hackintosh and all I did was just drop Lilu and CoreDisplayFixup in the appropriate kext folder in clover. For people who aren't using a Hackintosh, I'm not sure. I assume you'll have to put them in System/Library/Extensions and run an appropriate kext utility to get them to work.

**Comment by Floris497** [comment]
Exactly Lilu is the way to go now, it's a way more elegant way for this path :)

**Comment by pedrohex** [comment]
I dont have an hackintosh and i have some issues on external monitors. 
For instance apple mail when showing on second screen 

![screen shot 2017-10-09 at 21 29 56](https://user-images.githubusercontent.com/1288017/31357299-155c66d0-ad39-11e7-97e1-77259490e9bb.png)

**Comment by Floris497** [comment]
That does not look good at al.. no idea how to fix it :/

**Comment by pedrohex** [comment]
@Floris497 , i didnt install the patch. This is a clean high sierra on a mbp 2k11.

**Comment by pedrohex** [comment]
this means apple will be fixing some display issues next release.

**Comment by Floris497** [comment]
You're quite confident that Apple will do that next release.. please make sure you file a bug report at apple.

**Comment by pedrohex** [comment]
well... i tried but i dont have a developer account so...

**Comment by pedrohex** [comment]
ops.. got it : https://www.apple.com/feedback/macbookpro.html

**Comment by imguilherme** [comment]
Some update ?

**Comment by Floris497** [comment]
Not really.. no time.. look for lilu kext and so on :) should work

**Comment by Floris497** [comment]
Whoops

**Comment by juvenal** [comment]
@Floris497, there is an update: 10.13.1
Although I didn't see any fix on CoreDisplay, or any fix related to graphics output, did you have time to play a little with possible fixes for this patch?

**Comment by Floris497** [comment]
I'm sorry but I have no time at all right now, maybe in 2 weeks...

**Comment by TylerD89** [comment]
@juvenal Are you using a Hackintosh or real Mac? If you’re using a hackintosh, install Lilu.kext and CoreDisplayFixup.kext. It does the same thing but requires more work

**Comment by Floris497** [comment]
Both hack and Mac should be able to use lilu I think :)

**Comment by TylerD89** [comment]
Can you? Oh I thought SIP had to be disabled

**Comment by TylerD89** [comment]
Unfortunately not the case. CoreDisplayFixup needs to be injected into the firmware, it doesn’t work from /S/L/E

**Comment by Floris497** [comment]
Oh really.. wasn't aware of that.. also i have no clue how it's working.. if @PMheart would like to explain it to me he can send me an email or something :)

**Comment by TylerD89** [comment]
Yeah that’s beyond me. I have no idea how you people can code that stuff. I mean I know how to code, but I could never actually build something like that

**Comment by juvenal** [comment]
Hi @TylerD89, I'm using a Mac (two to be honest, one Mac mini late 2012 and a MacBook Pro late 2011). I updated the MacBook for High Sierra, but hold back the Mini with Sierra.
@Floris497, I tried to look at the CoreDisplay disassembly, but it seems to be completely rewritten, as the symbols are all different and in different places in the text section.
To better use my notebook with my desktop monitor, I'm looking for alternatives, like different kinds of connections between the machine and the monitor...
Anything that your experience could help?
Thanks for your time with this!

**Comment by Floris497** [comment]
I've already found the function to patch, it's also in the script but not activated.. the problem.. it breaks copy/past in the whole OS...

**Comment by TylerD89** [comment]
Yeah. Real Macs are kind of on hault right now... I mean you technically COULD install Clover and inject Fixup, but that will probably break your Mac...

**Comment by TylerD89** [comment]
I mean unless you want a $500 bill from AppleCare, be my guest lol

**Comment by Floris497** [comment]
Running clover on a real Mac is not that strange, it's done in some cases to run 64bit os on older macs that doe have 64bit CPU but no 64bit EFI.. so it should be safe

**Comment by rvprasad** [comment]
I was seeing blurry fonts on a 2560x1080@60Hz external display via HDMI and I fixed it by following [this advice](https://gist.github.com/adaugherity/7435890#gistcomment-2235540).

**Comment by spencerjason** [comment]
Hey @Bastani -- Any chance you have a more recent compile of Lilu + CoreDisplayFixup? 

The update of High Sierra to 10.13.2 is throwing a kernal panic with my current install.

**Comment by PMheart** [comment]
Hi! The discussions are too long to read! How can I be of service? :)

**Comment by Floris497** [comment]
Who is ready to test the new High Sierra patch?
I need a few testers :)

**Comment by pedrohex** [comment]
Hi Floris , how can I help ?

I’m using a MacBook Pro 13 early 2011 , Intel 3000 , i7 cpu.

On 26 Dec 2017, at 16:26, Floris Fredrikze <notifications@github.com<mailto:notifications@github.com>> wrote:


Who is ready to test the new High Sierra patch?
I need a few testers :)

**Comment by Floris497** [comment]
I've updated the patch. Could you check if it's working for you?

**Comment by jeffjacobsen** [comment]
MacMini (Late 2012)
10.13.2

Working again with LG Ultrawide (2560x1080) after patch

SIP looks to be disabled, all good!

(otool) Detected unpatched CoreDisplay on OS X 10.13.2 17C88.
SIP looks to be disabled, all good!
Patching CoreDisplay with patch version 4
Password:
Re-signing
/System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
/System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay:
replacing existing signature
Running 'sudo update_dyld_shared_cache' (Might need re-run after reboot) -
CAN TAKE A WHILE (Maybe even 10/30 minutes)
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/AssetCacheServicesExtensions.framework/Versions/A/XPCServices/AssetCacheTetheratorService.xpc/Contents/Frameworks/MobileDeviceKitLite.framework/Versions/A/MobileDeviceKitLite
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/BiometricKitUI.framework/Versions/A/BiometricKitUI
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/BookKit.framework/Versions/A/Frameworks/BookKitFoundation.framework/Versions/A/BookKitFoundation
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/BroadcomWLAN.framework/Versions/A/BroadcomWLAN
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/SetupAssistantSupport.framework/Versions/A/SetupAssistantSupport
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/Swift/libswiftDemangle.dylib
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/Swift/libswiftRemoteMirror.dylib
update_dyld_shared_cache: warning @rpath install name for system framework:
/System/Library/PrivateFrameworks/WirelessDiagnosticsSupport.framework/Versions/A/WirelessDiagnosticsSupport
update_dyld_shared_cache: i386 incorporating 626 OS dylibs, tracking 1047
others, building closures for 252 executables
update_dyld_shared_cache: x86_64 incorporating 1258 OS dylibs, tracking
1773 others, building closures for 2105 executables
update_dyld_shared_cache: warning: i386 adding dependents to
/System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant
Plugin.ibplugin/Contents/MacOS/Assistant Plugin: required dylib
'@rpath/InterfaceBuilderKit.framework/Versions/A/InterfaceBuilderKit' not
found, needed by
'/System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant
Plugin.ibplugin/Contents/MacOS/Assistant Plugin'
update_dyld_shared_cache: warning: x86_64 adding dependents to
/System/Library/QuickLook/Audio.qlgenerator/Contents/MacOS/Audio: required
dylib
'/Library/Frameworks/iTunesLibrary.framework/Versions/A/iTunesLibrary' not
found, needed by
'/System/Library/QuickLook/Audio.qlgenerator/Contents/MacOS/Audio'
Closing..

On Tue, Dec 26, 2017 at 12:35 PM, Floris Fredrikze <notifications@github.com

**Comment by Floris497** [comment]
How is copy paste doing? Still working?

**Comment by jeffjacobsen** [comment]
Everything seems fine - I haven't noticed any copy paste issues or any
other problems.

**Comment by Floris497** [comment]
Awesome!!!

**Comment by byicer** [comment]
@jeffjacobsen hello,can your macmini hold 4k@60hz?  I have a mac mini late 2012 too, i7 3720QM,HD4000,16G memory,and i want to use P2415Q or P2715Q with it.

**Comment by Floris497** [comment]
I think that screen can do only 30Hz and 60Hz for my mini 2011 it can go to 40Hz because my display supports refresrate in between 30 and 60.. and I think there is still MST SST problems with these screens from dell..

**Comment by byicer** [comment]
thank you～！

**Comment by tsl0922** [comment]
Works perfectly, Thank you!

**Comment by juvenal** [comment]
I just applied the patch on my MacBook Pro late 2011 with High Sierra 10.13.2, and my LG Ultrawide **(2560x1080)** came back to life (including system wide copy and paste).
I can confirm that the patch followed by 'sudo update_dyld_shared_cache' did the trick.
Everything is working again! Thanks a lot!

**Comment by Floris497** [comment]
Good to hear! Enjoy 😊

**Comment by Shasoosh** [comment]
Any idea if this will fix the fact that my High Sierra Hackintosh will not output 4K@60Hz through HDMI ?
